File: AllReactiveStackDemo/src/main/java/com/yxyl/webflux/web/DemoController.java
package com.yxyl.webflux.web;

import com.yxyl.webflux.entity.Post;
import com.yxyl.webflux.repository.FluxRepository;
import lombok.AllArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Flux;

import java.util.List;

/**
 * @program: WebFlux
 * @description: 测试Controller
 * @author: YxYL
 * @create: 2022-12-09 21:04
 **/
@RestController
@AllArgsConstructor
public class DemoController {

    private final FluxRepository fluxRepository;
    
    
    @GetMapping("/getAllByFlux")
    public Flux getALlByFlux() {
        Flux<Post> res = fluxRepository.findAll();
        return res;
    }
    
}

File: AllMVCStackDmeo/src/main/kotlin/com/yxyl/mvc/web/DemoController.kt
package com.yxyl.mvc.web

import com.yxyl.mvc.entity.Post
import com.yxyl.mvc.repository.MVCRepository
import lombok.AllArgsConstructor
import lombok.Data
import lombok.NoArgsConstructor
import lombok.NonNull
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController
import kotlin.reflect.KProperty

@RestController
@AllArgsConstructor
class DemoController() {

    @Autowired
    private lateinit var repository: MVCRepository

    @GetMapping("/getAllByMvc")
    fun getAll(): List<Post> = repository.findAll();

}

